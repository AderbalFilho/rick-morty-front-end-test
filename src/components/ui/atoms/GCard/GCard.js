import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './GCard.scss';

function GCard({ children, classes, variant, ...rest }) {
  const className = classNames('g-card', `g-card--${variant}`, classes);

  return (
    <div className={className} {...rest}>
      {children}
    </div>
  );
}

GCard.defaultProps = {
  variant: 'normal',
};

GCard.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
  variant: PropTypes.oneOf(['normal', 'loading', 'sheet', 'img-sheet']),
};

export default GCard;
