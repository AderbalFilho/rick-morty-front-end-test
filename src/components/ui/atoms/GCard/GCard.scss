@import 'styles/mixins';

.g-card {
  color: var(--color-white);
}

.g-card--normal {
  @include base-card;

  width: 100%;

  @media (min-width: 768px) {
    height: 224px;
    width: 224px;
  }
}

.g-card--loading {
  @include base-card;

  height: 187px;
  width: 187px;
}

.g-card--sheet {
  background-color: var(--color-black);
  border: 1px solid var(--color-light-black);
  border-radius: var(--radius-large);
  filter: var(--filter);
  margin: 0 12px;
  width: calc(100% - 24px);

  @media (min-width: 1440px) {
    height: 799px;
    margin: 0 auto;
    width: 1034px;
  }
}

.g-card--img-sheet {
  @include base-card;

  box-shadow: var(--shadow);
  height: 56%;

  @media (min-width: 1440px) {
    box-shadow: unset;
    height: 653px;
    width: 401px;
  }
}
