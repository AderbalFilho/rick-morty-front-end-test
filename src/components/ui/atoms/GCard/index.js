import GCard from './GCard';

export default GCard;
