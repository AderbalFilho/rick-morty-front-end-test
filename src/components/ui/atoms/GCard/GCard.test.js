import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import GCard from './GCard';

describe('GCard', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<GCard>inGaia</GCard>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  describe('change variant', () => {
    it('renders normal when no variant is passed', () => {
      const { container } = render(<GCard>inGaia</GCard>);
      const card = container.querySelector('.g-card');

      expect(card.classList).toContain('g-card--normal');
    });

    it('renders loading', () => {
      const { container } = render(<GCard variant="loading">inGaia</GCard>);
      const card = container.querySelector('.g-card');

      expect(card.classList).toContain('g-card--loading');
    });

    it('renders sheet', () => {
      const { container } = render(<GCard variant="sheet">inGaia</GCard>);
      const card = container.querySelector('.g-card');

      expect(card.classList).toContain('g-card--sheet');
    });

    it('renders img-sheet', () => {
      const { container } = render(<GCard variant="img-sheet">inGaia</GCard>);
      const card = container.querySelector('.g-card');

      expect(card.classList).toContain('g-card--img-sheet');
    });

    it('renders normal', () => {
      const { container } = render(<GCard variant="normal">inGaia</GCard>);
      const card = container.querySelector('.g-card');

      expect(card.classList).toContain('g-card--normal');
    });
  });

  it('renders children', () => {
    const { container } = render(<GCard>inGaia</GCard>);
    const card = container.querySelector('.g-card');

    expect(card.textContent).toBe('inGaia');
  });

  it('renders classes', () => {
    const { container } = render(<GCard classes="rick-card">inGaia</GCard>);
    const card = container.querySelector('.g-card');

    expect(card.classList).toContain('rick-card');
  });
});
