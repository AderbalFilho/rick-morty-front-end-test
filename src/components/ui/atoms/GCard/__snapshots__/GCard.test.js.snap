// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GCard matches snapshot 1`] = `
<div
  className="g-card g-card--normal"
>
  inGaia
</div>
`;
